File: app/src/main/java/com/amarnehsoft/holyquran/test/AyaHolder.java
package com.amarnehsoft.holyquran.test;

import android.media.MediaPlayer;

import com.amarnehsoft.holyquran.main.readersFragment.ReaderController;
import com.amarnehsoft.holyquran.model.Aya;

import java.io.IOException;

public class AyaHolder {

    private MediaPlayer mediaPlayer;
    private Aya aya;

    public AyaHolder(Aya aya, MediaPlayer.OnCompletionListener onCompletionListener, boolean withSound){
        this.aya = aya;
        if (withSound){
            mediaPlayer = new MediaPlayer();
            mediaPlayer.setOnCompletionListener(onCompletionListener);
            try {
                mediaPlayer.setDataSource(getSoundUrl(aya.getNumber()));
            } catch (IOException e) {
                e.printStackTrace();
            }

            try {
                mediaPlayer.prepare();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    private String getSoundUrl(int ayaNumber){
        return "http://cdn.alquran.cloud/media/audio/ayah/" + ReaderController.getCurrentReader().getId() + "/" + ayaNumber;
    }

    public Aya getAya() {
        return aya;
    }

    public void release(){
        if (mediaPlayer != null)
            mediaPlayer.release();
    }

    public boolean play() {
        if (mediaPlayer != null){
            mediaPlayer.start();
            return true;
        }
        return false;

    }

    public boolean pause(){
        if (mediaPlayer != null){
            mediaPlayer.pause();
            return true;
        }
        return false;

    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/test/AyaHolder.kt
package com.amarnehsoft.holyquran.test

import android.media.MediaPlayer

import com.amarnehsoft.holyquran.main.readersFragment.ReaderController
import com.amarnehsoft.holyquran.model.Aya

import java.io.IOException

class AyaHolder(
    val aya: Aya,
    onCompletionListener: MediaPlayer.OnCompletionListener,
    withSound: Boolean
) {

    private var mediaPlayer: MediaPlayer? = null

    init {
        if (withSound) {
            mediaPlayer = MediaPlayer()
            mediaPlayer?.setOnCompletionListener(onCompletionListener)
            try {
                mediaPlayer?.setDataSource(getSoundUrl(aya.number))
            } catch (e: IOException) {
                e.printStackTrace()
            }

            try {
                mediaPlayer?.prepare()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }

    private fun getSoundUrl(ayaNumber: Int): String {
        return "http://cdn.alquran.cloud/media/audio/ayah/" + ReaderController.currentReader.id + "/" + ayaNumber
    }

    fun release() = mediaPlayer?.release()

    fun play() = mediaPlayer?.start()?.let { true } ?: false

    fun pause() = mediaPlayer?.pause()?.let { true } ?: false
}
